app/src/main/kotlin/com/intive/picover/profile/viewmodel/ProfileViewModel.kt

```diff
@@ -73,15 +69,21 @@ class ProfileViewModel @Inject constructor(
 		}
 	}
 
-	fun validatingName(text: String) =
-		textValidator.validate(text)
+	fun onUsernameChange(username: String) {
+		_username.value = username
+	}
+
+	fun initUsername() {
+		_username.value = state.value.data().name
+	}
```

Comment:
This is something new I ques, why it's needed in `onEditNameClick`?

Reply:
It's needed to init username in the bottom sheet to be the same as showing in the profile so that we can update it. Without this initialization, the text field will have previously typed valued

Reply:
So it's seems that we faced first issue with using the same view model for `ProfileScreen` and bottom sheets, which should rather be independent from `ProfileScreen`. That will be more clear when we navigate to bottom sheets by navigation component instead of current "if" approach. Let's keep it as it is but please place a TODO that it needs to be changed. 
It's a nice example of potential cause for future bugs, because entering the bottom sheet for subsequential time depends on the previous one, which should avoid that.

Reply:
Probably the same thing can happen in `PartiesScreen` and its bottom sheet

Reply:
TODO placed

Reply:
Seems to be too generic. Rest of the team probably wouldn't what to do. Please place a TODO next to `initUsername` call in `onEditNameClick`. The explanation should contain a clear reason for it, in this case the clue of _"Without this initialization, the text field will have previously typed valued"_

Reply:
Update

Reply:
Perfect!